### PR TITLE
[TASK] Feature: Enable company to manage its users and enable platform admins to manage all users

### PR DESCRIPTION
**Pull request by Leonardomontesqui**
Pr time!!!!! just needs the supabase service_role key :)

### CODE REVIEW

**Comment by marcelo-cm** [comment]
image

Is this supposed to be here? if so, it looks bad

**Review by marcelo-cm — CHANGES_REQUESTED** [review]
wazzaaapp

File: lib/hooks/supabase/useUser.ts [hunk 1]
```diff
@@ -119,13 +156,67 @@ export const useUser = () => {
...
+    return;
+  };
+
+  const deleteUserDB = async (userID: UUID) => {
+    console.log('this is the userID that got deleted: ', userID);
+    const { data: userData, error: userError } = await fetchUser();
+
+    if (userError) {
+      console.log(
+        'Yo this is the errror when fetchin session user: ',
```
[the remark this answers is not in the document]
Comment (Leonardomontesqui) [reply lib/hooks/supabase/useUser.ts:178]: Looool this was from when i was trynna figure out why I was getting logged out after creating a user

File: lib/hooks/supabase/useCompany.ts [hunk 2]
```diff
@@ -5,31 +5,29 @@ import { User_Nested } from '@/models/User';
...
     if (error) {
+      console.log('there was an error creating the company ', error);
       throw new Error('Failed to create company');
     }
 
     return data;
   };
 
-  const fetchCompanyData = async (
-    columns: (keyof Company)[] | ['*'] = ['*'],
```
[the remark this answers is not in the document]
Comment (Leonardomontesqui) [reply lib/hooks/supabase/useCompany.ts:25]: yea, im fetching all

File: components/(dashboards)/teams/AddUserForm.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        <div className="flex items-center gap-2">
+          <p className="w-[90px] text-base">Roles</p>
+
+          <MultiComboBox
+            className="w-full"
+            options={[
+              { id: 'PLATFORM_ADMIN' },
+              { id: 'COMPANY_ADMIN' },
+              { id: 'BOOKKEEPER' },
+            ]}
```
[the remark this answers is not in the document]
Comment (Leonardomontesqui) [reply components/(dashboards)/teams/AddUserForm.tsx:156]: Object.values(Roles).map((role) => ({ id: role }))

**Comment by Leonardomontesqui** [comment]
New PR

**Review by marcelo-cm — APPROVED** [review]
can go in once these important small changes r done.

**Pull request merged**